test_conformance/api/negative_queue.cpp

```diff
@@ -57,55 +151,107 @@ int test_negative_create_command_queue_with_properties(cl_device_id deviceID,
-        context, deviceID, queue_prop_def, &test_error);
-
-    test_failure_error_ret(test_error, CL_INVALID_QUEUE_PROPERTIES,
-                           "clCreateCommandQueueWithProperties should "
-                           "return CL_INVALID_QUEUE_PROPERTIES if "
-                           "values specified in properties are valid but "
-                           "are not supported by the "
-                           "device.",
-                           TEST_FAIL);
+    cl_command_queue_properties device_queue_properties = 0;
+    err = clGetDeviceInfo(deviceID, CL_DEVICE_QUEUE_PROPERTIES,
+                          sizeof(device_queue_properties),
+                          &device_queue_properties, nullptr);
+    test_error(err, "clGetDeviceInfo");
+    std::initializer_list<cl_command_queue_properties> valid_properties{
+        CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE, CL_QUEUE_PROFILING_ENABLE
+    };
+    properties[0] = CL_QUEUE_PROPERTIES;
+    bool missing_property = false;
+    // Iterate through all possible properties to find one which isn't supported
+    for (auto property : valid_properties)
+    {
+        if ((device_queue_properties & property) == 0)
+        {
+            missing_property = true;
+            properties[1] = property;
+            break;
+        }
+    }
+    if (missing_property)
+    {
+        clCreateCommandQueueWithProperties(context, deviceID, properties.data(),
+                                           &err);
+        test_failure_error_ret(
+            err, CL_INVALID_QUEUE_PROPERTIES,
+            "clCreateCommandQueueWithProperties should return "
+            "CL_INVALID_QUEUE_PROPERTIES when: \"values specified in "
+            "properties are valid but are not supported by the device\"",
+            TEST_FAIL);
+    }
+    else if (get_device_cl_version(deviceID) >= Version(2, 0))
+    {
+        cl_uint max_size = -1;
+        err = clGetDeviceInfo(deviceID, CL_DEVICE_QUEUE_ON_DEVICE_MAX_SIZE,
+                              sizeof(max_size), &max_size, nullptr);
+        test_error(err, "clGetDeviceInfo");
+        if (max_size > 0)
+        {
+            properties[0] = CL_QUEUE_PROPERTIES;
+            properties[1] =
+                CL_QUEUE_ON_DEVICE | CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE;
+            properties[2] = CL_QUEUE_SIZE;
+            properties[3] = max_size + 1;
+            clCreateCommandQueueWithProperties(context, deviceID,
+                                               properties.data(), &err);
+            if (err != CL_INVALID_VALUE && err != CL_INVALID_QUEUE_PROPERTIES)
+            {
+                log_error(
+                    "clCreateCommandQueueWithProperties should return "
+                    "CL_INVALID_VALUE or CL_INVALID_QUEUE_PROPERTIES when: "
+                    "\"values specified in properties are not valid\" using a "
+                    "queue size greather than "
+                    "CL_DEVICE_QUEUE_ON_DEVICE_MAX_SIZE");
```

Comment:
If error occurs, it prints 
negative_create_command_queue_with_properties...
clCreateCommandQueueWithProperties should return CL_INVALID_VALUE or CL_INVALID_QUEUE_PROPERTIES when: "values specified in properties are not valid" using a queue size greather than CL_DEVICE_QUEUE_ON_DEVICE_MAX_SIZEnegative_create_command_queue_with_properties FAILED
                                                                    
can we add "\n" or "." to split the 2 sentences? Same comments for other log_error call.